src/transaction/components/TransactionButton.tsx

```diff
@@ -19,10 +19,9 @@ export function TransactionButton({
     chainId,
     errorMessage,
     isLoading,
+    lifeCycleStatus,
     onSubmit,
     receipt,
-    statusWriteContract,
-    statusWriteContracts,
```

Comment:
Instead of passing data back and forth from random components, we align the way data is stream by having:
- from the parent pushing down status update
- from the child to the parent set the status

So ideally `lifeCycleStatus` directs the experience of all components under the same provider.

This creates a [drum bit](https://www.youtube.com/watch?v=gUQG8CL2rfY), that help set the tone across all components.

cc @abcrane123 @0xAlec @cpcramer